### PR TITLE
Add support for the secrets bundle from MSC4108

### PR DESCRIPTION
**Pull request by poljar**

### CODE REVIEW

File: crates/matrix-sdk-crypto/src/store/mod.rs [hunk 1]
```diff
@@ -1214,6 +1264,102 @@ impl Store {
...
+        Ok(SecretsBundle {
+            cross_signing: CrossSigningSecrets { master_key, user_signing_key, self_signing_key },
+            backup,
+        })
+    }
+
+    /// Import and persists secrets from a [`SecretsBundle`].
+    ///
+    /// This method will import all the private cross-signing keys and, if
+    /// available, the private part of a backup key and it's accompanying
```
Comment (Hywan) [inline crates/matrix-sdk-crypto/src/store/mod.rs:1318]:
```diff
- it's
+ its
```

**Review by Hywan — APPROVED** [review]
Minor feedback; great code as usual, thanks!

File: crates/matrix-sdk-crypto/src/store/mod.rs [hunk 2]
```diff
@@ -1214,6 +1264,102 @@ impl Store {
...
+    /// Export all the secrets we have in the store into a [`SecretsBundle`].
+    ///
+    /// This method will export all the private cross-signing keys and, if
+    /// available, the private part of a backup key and it's accompanying
+    /// version.
+    ///
+    /// The method will fail if we don't have all three private cross-signing
+    /// keys available.
+    ///
+    /// **Warning**: Only export this and share it with a trusted reciepient,
```
[the remark this answers is not in the document]
Comment (Hywan) [reply crates/matrix-sdk-crypto/src/store/mod.rs:1276]: I am surprised the CI didn't catch this typo.

[the remark this answers is not in the document]
Comment (poljar) [reply crates/matrix-sdk-crypto/src/store/mod.rs]: Gotta add it to `typos`: https://github.com/crate-ci/typos/issues/1007#issuecomment-2109644928

**Pull request merged**